lib/resources/user-collection.js

```diff
@@ -155,7 +155,8 @@ UserCollection.prototype.handle = function (ctx) {
       if(ctx.query.id || ctx.body.id) {
         this.save(ctx, done);
       } else {
-        this.store.first({username: ctx.body.username}, function (err, u) {
+        var usernameRegex = new RegExp(["^", ctx.body.username, "$"].join(""), "i");
```

Comment:
There is a problem here if ctx.body.username contains regex control characters. The input would need to be escaped for this to work properly, otherwise the username can't contain a dot, $, ^. [ and a bunch of other characters. 

Using a regex here also hurts performance, since an index can't be used.

I think the solution is more delicate than this, an internal `username_lowercase` field should be populated/used instead.